pig/src/main/java/org/apache/zeppelin/pig/PigInterpreter.java

```diff
@@ -0,0 +1,138 @@
+    if (includeJobStats != null) {
+      this.includeJobStats = Boolean.parseBoolean(includeJobStats);
+    }
+    try {
+      pigServer = new PigServer(execType);
+    } catch (IOException e) {
+      throw new RuntimeException("Fail to launch PigServer", e);
+    }
+  }
+
+  @Override
+  public void close() {
+    pigServer = null;
+  }
+
+
+  @Override
+  public InterpreterResult interpret(String cmd, InterpreterContext contextInterpreter) {
+    // remember the origial stdout, because we will redirect stdout to capture
+    // the pig dump output.
+    PrintStream originalStdOut = System.out;
+    ByteArrayOutputStream bytesOutput = new ByteArrayOutputStream();
+    File tmpFile = null;
+    try {
+      tmpFile = PigUtils.createTempPigScript(cmd);
+      System.setOut(new PrintStream(bytesOutput));
+      // each thread should its own ScriptState & PigStats
+      ScriptState.start(pigServer.getPigContext().getExecutionEngine().instantiateScriptState());
+      // reset PigStats, otherwise you may get the PigStats of last job in the same thread
+      // because PigStats is ThreadLocal variable
+      PigStats.start(pigServer.getPigContext().getExecutionEngine().instantiatePigStats());
+      PigScriptListener scriptListener = new PigScriptListener();
+      ScriptState.get().registerListener(scriptListener);
+      listenerMap.put(contextInterpreter.getParagraphId(), scriptListener);
+      pigServer.registerScript(tmpFile.getAbsolutePath());
+    } catch (IOException e) {
+      if (e instanceof FrontendException) {
+        FrontendException fe = (FrontendException) e;
+        if (!fe.getMessage().contains("Backend error :")) {
+          // If the error message contains "Backend error :", that means the exception is from
+          // backend.
+          LOGGER.error("Fail to run pig script.", e);
+          return new InterpreterResult(Code.ERROR, ExceptionUtils.getStackTrace(e));
+        }
+      }
+      PigStats stats = PigStats.get();
+      if (stats != null) {
+        String errorMsg = PigUtils.extactJobStats(stats);
+        if (errorMsg != null) {
+          LOGGER.error("Fail to run pig script, " + errorMsg);
+          return new InterpreterResult(Code.ERROR, errorMsg);
+        }
+      }
+      return new InterpreterResult(Code.ERROR, ExceptionUtils.getStackTrace(e));
+    } finally {
+      System.setOut(originalStdOut);
+      listenerMap.remove(contextInterpreter.getParagraphId());
+      if (tmpFile != null) {
+        tmpFile.delete();
+      }
+    }
+    StringBuilder outputBuilder = new StringBuilder();
+    PigStats stats = PigStats.get();
+    if (stats != null && includeJobStats) {
+      String jobStats = PigUtils.extactJobStats(stats);
+      if (jobStats != null) {
+        outputBuilder.append(jobStats);
+      }
+    }
+    if (!outputBuilder.toString().isEmpty() || !bytesOutput.toString().isEmpty()) {
+      outputBuilder.append("------------- Pig Output --------------\n");
```

Comment:
I'm not sure we have a "header" in other interpreter output - is there a reason this could be useful?